### PR TITLE
Head Node enables root login and ssh on other nodes

### PR DESCRIPTION
**Pull request by whoarethebritons**
Additionally compatibility is checked on just the head node right now
https://ocd.appscale.com:8080/job/Daily%20Build/2289/
https://ocd.appscale.com:8080/view/All/job/Tools-test/1169/console

### CODE REVIEW

File: appscale/tools/remote_helper.py [hunk 1]
```diff
@@ -108,6 +108,31 @@ def start_all_nodes(cls, options, count):
 
     return instance_ids, public_ips, private_ips
 
+  @classmethod
+  def enable_head_node_root_ssh(cls, options, head_node):
```
Comment (menivaitsi) [inline appscale/tools/remote_helper.py]: Please rename to: `def enable_root_ssh`

File: appscale/tools/remote_helper.py [hunk 2]
```diff
@@ -108,6 +108,31 @@ def start_all_nodes(cls, options, count):
...
+    """Enables root logins on the head node, enables SSH access,
+    and copies the user's SSH key to the head node.
+
+    Args:
+      options: A Namespace that specifies the cloud infrastructure to use, as
+        well as how to interact with that cloud.
+      head_node: The head_node from the NodeLayout.
+    """
+    cls.sleep_until_port_is_open(head_node.public_ip, cls.SSH_PORT,
```
Comment (menivaitsi) [inline appscale/tools/remote_helper.py]: Just pass in the public_ip - no need for the entire head_node object.

File: appscale/tools/remote_helper.py [hunk 3]
```diff
@@ -108,6 +108,31 @@ def start_all_nodes(cls, options, count):
...
+                                 options.verbose)
+
+    # Since GCE v1beta15, SSH keys don't immediately get injected to newly
+    # spawned VMs. It takes around 30 seconds, so sleep a bit longer to be
+    # sure.
+    if options.infrastructure == 'gce':
```
Comment (menivaitsi) [inline appscale/tools/remote_helper.py]: This check is irrelevant in this context. I think it should be done before this method.

File: appscale/tools/remote_helper.py [hunk 4]
```diff
@@ -108,6 +108,31 @@ def start_all_nodes(cls, options, count):
...
+      AppScaleLogger.log("Waiting for SSH keys to get injected in your "
+        "machine(s).")
+      time.sleep(60)
+
+    cls.enable_root_login(head_node.public_ip, options.keyname,
+      options.infrastructure, options.verbose)
```
Comment (menivaitsi) [inline appscale/tools/remote_helper.py]: Align arguments vertically.

File: appscale/tools/appscale_tools.py [hunk 5]
```diff
@@ -508,13 +509,16 @@ def run_instances(cls, options):
         node_layout.nodes[i].public_ip = public_ips[i]
         node_layout.nodes[i].private_ip = private_ips[i]
         node_layout.nodes[i].instance_id = instance_ids[i]
+      #enable root ssh & login on head node
```
Comment (menivaitsi) [inline appscale/tools/appscale_tools.py]: Allow one whitespace after # and form full sentences in comments. 
Same below.

**Comment by whoarethebritons** [comment]
https://ocd.appscale.com:8080/view/All/job/Daily%20Build/2306/

File: appscale/tools/remote_helper.py [hunk 6]
```diff
@@ -108,6 +108,24 @@ def start_all_nodes(cls, options, count):
 
     return instance_ids, public_ips, private_ips
 
+  @classmethod
+  def enable_root_ssh(cls, options, public_ip):
+    """Enables root logins on the machine, enables SSH access,
```
Comment (tmarballi) [inline appscale/tools/remote_helper.py]: Could you change this sentence to -> Enables root logins and SSH access on the machine

File: appscale/tools/appscale_tools.py [hunk 7]
```diff
@@ -509,13 +510,18 @@ def run_instances(cls, options):
         node_layout.nodes[i].public_ip = public_ips[i]
         node_layout.nodes[i].private_ip = private_ips[i]
         node_layout.nodes[i].instance_id = instance_ids[i]
+
+      #Enable root ssh & login on the head node.
+
```
Comment (tmarballi) [inline appscale/tools/appscale_tools.py]: Can you please remove the extra empty line here?

**Pull request merged**